install_mac.py

```diff
@@ -145,73 +152,96 @@ def install_brew_packages(disable: List[str]):
 
-    python3('-m', 'pip', 'install', '-U', 'pip')
+    confirm_install('upgrade pip', python3['-m', 'pip', 'install', '-U', 'pip'])
+
+    python_packages = ['xattr', 'pyfzf', 'artifactory', 'humanfriendly', 'pygments', 'ipython', 'plumbum',
+                       'pymobiledevice3', 'harlogger', 'cfprefsmon', 'pychangelog2']
 
-    python3('-m', 'pip', 'install', '-U', 'xattr', 'xonsh', 'pyfzf', 'artifactory', 'humanfriendly', 'pygments',
-            'ipython', 'plumbum', 'xontrib-argcomplete', 'xontrib-fzf-widgets', 'xontrib-z', 'xontrib-up',
-            'xontrib-vox', 'xontrib-jedi', 'pymobiledevice3', 'harlogger', 'cfprefsmon', 'pychangelog2')
+    for package in python_packages:
+        confirm_install(f'install {package}', python3['-m', 'pip', 'install', '-U', package])
 
 
 def install_xonsh():
     logger.info('installing xonsh')
 
+    confirm_install('upgrade pip', python3['-m', 'pip', 'install', '-U', 'pip'])
+
+    python3('-m', 'pip', 'install', '-U', 'xonsh')
+
+    confirm_install(f'install xonsh attributes', python3['-m', 'pip', 'install', '-U', 'xontrib-argcomplete',
+                    'xontrib-fzf-widgets', 'xontrib-z', 'xontrib-up', 'xontrib-vox', 'xontrib-jedi'])
+
     xonsh_path = shutil.which('xonsh')
     if xonsh_path not in open('/etc/shells', 'r').read():
         sudo('sh', '-c', f'echo {xonsh_path} >> /etc/shells')
 
-    brew('reinstall', 'fzf')
-    brew('reinstall', 'bash-completion')
+    confirm_install('install/reinstall fzf', brew['reinstall', 'fzf'])
+    confirm_install('install/reinstall bash-completion', brew['reinstall', 'bash-completion'])
 
-    DEV_PATH.mkdir(parents=True, exist_ok=True)
+    confirm_install('set xonsh to be the default shell', chsh['-s', xonsh_path])
 
-    os.chdir(DEV_PATH)
-    git_clone('git@github.com:doronz88/worksetup.git', 'main')
-    cp('worksetup/.xonshrc', Path('~/').expanduser())
-    chsh('-s', xonsh_path)
+    def set_xonshrc():
+        DEV_PATH.mkdir(parents=True, exist_ok=True)
 
+        os.chdir(DEV_PATH)
+        git_clone('git@github.com:doronz88/worksetup.git', 'main')
+        cp('worksetup/.xonshrc', Path('~/').expanduser())
 
-@click.group()
-def cli():
-    pass
+    confirm_install('set ready-made .xonshrc file', set_xonshrc)
 
 
-@cli.command('configure-preferences')
-@click.option('-a', '--automated', is_flag=True, help='do everything without prompting')
-def cli_configure_preferences(automated):
-    if automated:
+def set_automation(ctx, param, value):
+    if value:
         global AUTOMATION_MODE
         AUTOMATION_MODE = True
+
+
+class BaseCommand(click.Command):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.params[:0] = [
+            click.Option(('-a', '--automated'), is_flag=True, callback=set_automation, expose_value=False,
+                         help='do everything without prompting')]
+
+
+@click.group(help='use -a/--automated flag for doing every command without prompts')
```

Comment:
- use docstring for help message instead
- this should doc the command and not a specific option. for that we'll have the specific option's own help

Reply:
I wanted to do that if anyone call the group (`python3 install_mac.py`) without call a specific subcommand he will see the `automated` option also. And i did not find out how to do so :(

Reply:
anyway, i remove it now...